projects/step-frontend/src/lib/modules/timeseries/modules/legacy/components/time-series-dashboard/time-series-dashboard.component.ts

```diff
@@ -102,6 +108,12 @@ export class TimeSeriesDashboardComponent implements OnInit, OnDestroy {
     this.contextInitialized.next(this.context);
   }
 
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes['timeRange']?.currentValue) {
+      this.handleTimeRangeChange({ selection: changes['timeRange'].currentValue, triggerRefresh: true });
+    }
```

Comment:
It's better to check with previous value, to avoid some extra calls in case if value is the same

```suggestion
    const cTimeRange = changes['timeRange'];
    if (!!cTimeRange?.currentValue && (cTimeRange?.currentValue !== cTimeRange?.previousValue || cTimeRange?.firstChange)) {
      this.handleTimeRangeChange({ selection: cTimeRange.currentValue, triggerRefresh: true });
    }
```